File: 이건희/level1/[1차] 캐시.py
def solution(cacheSize, cities):
    result = 0
    cache = []
    
    for i in cities:
        i = i.lower()
        if not i in cache:
            result += 5
            if len(cache) < cacheSize:
                cache.append(i)
            else:
                cache.append(i)
                cache.pop(0)
        else:
            result += 1
            cache.remove(i)
            cache.append(i)
            
    return result

File: 이훈일/Programmers/Level2/[1차]캐시.java
import java.util.*;

class Solution {
    public int solution(int cacheSize, String[] cities) {
        int result = 0;
        ArrayList<String> cache = new ArrayList<>();

        if(cacheSize == 0) {
            return cities.length * 5;
        }

        for(int i = 0;i < cities.length;i++) {
            String city = cities[i].toUpperCase();

            if(cache.contains(city)) {
                cache.remove(city); 
                cache.add(city); 
                result += 1;
            } else { 
                if(cache.size() == cacheSize) {
                    cache.remove(0); 
                }
                
                cache.add(city); 
                result += 5;
            }
        }

        return result;
    }
}
